src/runtime_src/core/edge/drm/zocl/sched_exec.c

```diff
@@ -2473,11 +2547,13 @@ ps_ert_query(struct sched_cmd *cmd)
 		__attribute__ ((fallthrough));
 		/* pass through */
 
+	case ERT_CU_STAT:
 	case ERT_CONFIGURE:
 		mark_cmd_complete(cmd, ERT_CMD_STATE_COMPLETED);
 		break;
 
 	default:
+		mark_cmd_complete(cmd, ERT_CMD_STATE_ERROR);
```

Comment:
For an unrecognized command, we probably want take care of it in penguin_submit and ps_ert_submit as well to mark submit error. Looks like then won't be queried here if they are not known command.

Reply:
I think so. I would like to handle the unrecognized command at cq_check(). But I found I'll need to maintain a supported command list at cq_check() and maintain the consistency with query and submit.
The submit() could return "false" to just skip the unknown opcode. We need more changes to support submit error in current KDS implementation. 
This is the easiest way I can find to handle unknown opcode. Let's refine this at KDS/ERT 3.0.

Reply:
Can we just call mark_cmd_submit_error() in ps_ert_submit and penguin_submit()? But that could be a different PR.